Source/DafnyCore/AST/Members/Function.cs

```diff
@@ -65,15 +65,14 @@ public class Function : MemberDecl, TypeParameter.ParentType, ICallable, ICanFor
       yield return new Assumption(this, tok, AssumptionDescription.NoBody(IsGhost));
     }
 
-    if (HasExternAttribute) {
+    if (HasExternAttribute && !HasAxiomAttribute) {
```

Comment:
I believe the auditor wants to report on everything that has `{:axiom}` on it. The way I understood the auditor is that you don't run it as part of CI, but use it as part of a whole-program code review that you do at particular times in your development workflow, like before a release or when onboarding a new customer.

Reply:
Yes, that's my intention for it as well. It's for when you want to say "this thing is fully verified, but what risks remain?". For that, we really want to keep all extern declarations in the list of things reported by the auditor.

I think this approach is a good one for silencing warnings during the normal development process. In the context of the auditor, I think having `{:axiom}` should convert an extern declaration from an implicit assumption to an explicit assumption. Explicit assumptions are those that have been acknowledged as assumptions through some extra annotation and are therefore considered intentional. The final goal for a verified project is, in my mind, for all remaining assumptions to be explicit, and to have documented reasons for why you think that each one is warranted.

Reply:
On thing that might not be obvious here: the state before this change is that `function {:extern} {:axiom} ...` would always produce two Assumptions: one for the presence of `{:axiom}` in general, and one for the fact that you're assuming that an extern actually behaves like a non-deterministic function. The first is considered an explicit assumption and the second an implicit one. The intent is that doo files only allow explicit assumptions, so the first was allowed but the second caused an error.

It was already true that adding `{:axiom}` to an extern method or function would replace the implicit assumptions for any post-conditions by an explicit one just saying "you have an axiom here" instead. The effect of this change is just to let the explicit `{:axiom}` assumption suppress the additional implicit one about the function-ness of your extern too. Hence just adding `&& !HasAxiomAttribute` makes sense for both the auditor and building doo files.

Reply:
Makes sense. I had trouble seeing that the explicit assumption is still flagged.

Reply:
Ditto. I agree in that broader context.